modules/core/src/lib/effect.ts

```diff
@@ -11,7 +11,9 @@ export type PostRenderOptions = LayersPassRenderOptions & {
 export interface Effect {
   id: string;
   props: any;
+  useInCollide?: boolean;
   useInPicking?: boolean;
+  didRender?: boolean;
```

Comment:
This seems weirdly specific. What if the Effect does not render anything? Maybe something like `get isEnabled(): boolean` more clearly reflects that it is a state of the instance?

Reply:
The whole point is to detect if something was rendered. In this case if the mask map is updated, as this requires the collide map to be redrawn. 

Another option here would be to add some state to the mask map (e.g. `MaskEffect.maskMap.userData.version`) which `MaskPass.render` would increment, but I prefer to keep it at the effect level.

Reply:
My worry is that this field is essentially a dirty flag, but there is no mechanism to clear it. We have like 5 different effects now, and we shouldn't add a new flag then leave it up to each effect class to manage.

May I propose the following change:

- `Effect.preRender(options)` can return an arbitrary value
- The render pipeline assembles these return values and pass it downstream, i.e. `options.preRenderStats["mask-effect"] = effect.preRender(options)`
- `CollideEffect.preRender` can then look at `options.preRenderStats` to see the preRender result of its dependencies

Would that solve your problem?